integrations/anthropic/src/haystack_integrations/components/generators/anthropic/generator.py

```diff
@@ -126,6 +147,12 @@ def run(self, prompt: str, generation_kwargs: Optional[Dict[str, Any]] = None):
          - `replies`: A list of generated replies.
          - `meta`: A list of metadata dictionaries for each reply.
         """
+
+        # Select the model adapter based on the provided env variables
+        model_adapter_cls = self.get_model_adapter()
+        client = model_adapter_cls.client()
+        model_name = model_adapter_cls.set_model(self.model)  # set the model name in the format required by the API
```

Comment:
I would prefer to initialize the client locally if it doesnt have any implications.